src/satosa/satosa_config.py

```diff
@@ -30,11 +31,12 @@ def __init__(self, config):
         :param config: Can be a file path or a dictionary
         :return: A verified SATOSAConfig
         """
-        parsers = [self._load_dict, self._load_yaml]
-        for parser in parsers:
-            self._config = parser(config)
-            if self._config is not None:
-                break
+        self._config = self._parse_config(config)
+
+        if not self._config:
+            raise SATOSAConfigurationError(
+                "Missing configuration or unknown format"
```

Comment:
How can we know which of the two it is?

Reply:
We didn't and we don't

Reply:
How can we provide this information?

Reply:
Uknown format means parsing failed or returned empty dict whereas missing configuration could mean that either an empty config argument was given to the constructor or that the argument was a file that couldn't be opened. The first case was/is only indirectly checked. The second will be displayed in the logs (if `_load_yaml` cannot open the file for example). I'm not saying that this is the way it should be done, I'm just saying how it is currently implemented

Reply:
> I'm not saying that this is the way it should be done, I'm just saying how it is currently implemented

Yes, I know; and what I'm saying is that we should make clear what went wrong. How we make it clearer is what we are discussing.

Ideas to improve this:

- Provide `config` as part of the message and let the user figure out which of the two it is implicitly
- Make more fine grained checks and provide the exact message

Reply:
These should probably be written in a new issue, otherwise in a PR they will be lost